third_party/qsort_arg.c

```diff
@@ -197,20 +198,39 @@ loop:SWAPINIT(a, es);
 		pc -= es;
 	}
 	pn = (char *) a + n * es;
-	r = min(pa - (char *) a, pb - pa);
-	vecswap(a, pb - r, r);
-	r = min(pd - pc, pn - pd - (intptr_t)es);
-	vecswap(pb, pn - r, r);
-	if ((r = pb - pa) > (intptr_t)es)
-		qsort_arg(a, r / es, es, cmp, arg);
-	if ((r = pd - pc) > (intptr_t)es)
+	d1 = min(pa - (char *) a, pb - pa);
+	vecswap(a, pb - d1, d1);
+	d1 = min(pd - pc, pn - pd - es);
+	vecswap(pb, pn - d1, d1);
+	d1 = pb - pa;
+	d2 = pd - pc;
+	if (d1 <= d2)
 	{
-		/* Iterate rather than recurse to save stack space */
-		a = pn - r;
-		n = r / es;
-		goto loop;
+		/* Recurse on left partition, then iterate on right partition */
+		if (d1 > es)
+			qsort_arg(a, d1 / es, es, cmp, arg);
```

Comment:
Nit: actually qsort_arg_st must be here.

Reply:
Right. Fixed.